src/crate/client/cursor.py

```diff
@@ -306,12 +300,17 @@ def _to_datetime_with_tz(value: t.Optional[float]) -> t.Optional[datetime]:
     @staticmethod
     def _timezone_from_utc_offset(tz) -> timezone:
         """
-        Convert UTC offset in string format (e.g. `+0530`) into `datetime.timezone` object.
+        UTC offset in string format (e.g. `+0530`) to `datetime.timezone`.
         """
-        assert len(tz) == 5, f"Time zone '{tz}' is given in invalid UTC offset format"
+        # TODO: Remove use of `assert`. Better use exceptions?
```

Comment:
Only if you will catch the exception where you call `_timezone_from_utc_offset`

Reply:
On this spot, I am referring to the fact that the ruff linter discourages the use of `assert` statements in code:
```
S101 Use of `assert` detected
```

That's why it needs to be suppressed now, using `# noqa: S101`.

My proposal would be to rewrite the `assert` statements to constructs like that:
```python
if len(tz) != 5:
    raise ValueError(f"Time zone '{tz}' is given in invalid UTC offset format")
```

I think that would be the correct approach to modernize the code base. wdyt?

Reply:
Now it makes sense, and I agree that raising an exception is a better option.